### PR TITLE
Refacter/serverlist

### PR DESCRIPTION
**Pull request by ZhuYuxuan9302**

### CODE REVIEW

**Comment by MCJPG-Agent** [comment]
🎉 感谢 @ZhuYuxuan9302 提交 Pull Request!
🎉 Thanks @ZhuYuxuan9302 for submitting the Pull Request!

我们会尽快审查您的代码。在此期间,请确保:
We will review your code as soon as possible. In the meantime, please ensure:
- ✅ 所有测试通过 / All tests pass
- ✅ 代码符合项目规范 / Code follows project conventions
- ✅ 已添加必要的文档 / Necessary documentation is added

🤖 我们的 **MCJPG Agent** 将自动对您的代码进行初步审查。
🤖 Our **MCJPG Agent** will automatically conduct an initial review of your code.

如有代码辅助需求,请在评论区 `@claude` 进行请求。
If you need code assistance, please `@claude` in the comments.

**Comment by MCJPG-Agent** [comment]
**Claude finished @ZhuYuxuan9302's task in 1m 57s** —— [View job](https://github.com/MineJPGcraft/MCJPG/actions/runs/27735480468)

---
### PR #149 代码审查 

- [x] 获取 PR 上下文
- [x] 读取并分析变更文件
- [x] 审查 `CLAUDE.md` 变更
- [x] 审查 `ServerList.vue` 变更
- [x] 总结审查结果

---

### 总体评价

此 PR 对 `ServerList.vue` 进行了较大规模的重构，将原先的卡片式链接改为**可展开的手风琴式交互**，并新增了图片画廊、IP 复制和 `medium-zoom` 图片放大功能，交互设计整体上有明显改善。同时 `CLAUDE.md` 被重写为完整的开发指南，信息量大幅提升。

---

### `CLAUDE.md` 变更

**优点：**
- 从简短说明扩展为详尽的开发指南，包含目录结构、i18n 架构、组件列表、开发命令等，对开发者很有价值。

**问题：**

1. **新增了 Giscus、Algolia 等敏感配置信息**（第 238-253 行区域）：
   - Algolia 的 `appId`、`indexName`、AI search assistant ID 等已硬编码在 `CLAUDE.md` 中。虽然 Algolia 的公开搜索 key 通常只用于只读操作、危险性有限，但将这类标识符放入 CLAUDE.md（AI 可读取的文档）仍需评估风险。
   - Giscus 的 `repoId`、`categoryId` 也被公开记录。
   - **建议**：确认这些信息是否已在其他地方公开（如源码中），如果已公开则影响较小；如果是首次公开暴露，建议移除或用占位符替代。

2. **目录结构描述与实际目录可能不一致**：文档中写的是 `configs/` 但实际代码是 `docs/.vitepress/config/`（原 CLAUDE.md 中写的是 `/docs/.vitepress/config`）。建议核实目录名称确保准确。

---

### `ServerList.vue` 变更

#### 🐛 Bug / 功能问题

1. **`medium-zoom` 在 `onMounted` 中未绑定元素选择器，可能导致初始图片无法放大** ([`ServerList.vue:766`](https://github.com/MineJPGcraft/MCJPG/blob/refacter/serverlist/docs/.vitepress/theme/components/ServerList.vue#L766))：
   ```js
   zoomInstance = mediumZoom({ background: 'var(--vp-c-bg)' })
   ```
   初始化时没有传入选择器，依赖后续 `watch` 的 `attach`。但初始加载完成后，如果 `paginatedServers` 或 `expandedIds` 没有变化，`watch` 不会触发，首批展开的图片可能无法使用 zoom。建议改为：
   ```js
   zoomInstance = mediumZoom('.gallery-img', { background: 'var(--vp-c-bg)' })
   ```
   或在数据加载完成后调用一次 `attach`。[Fix this →](https://claude.ai/code?q=In%20MineJPGcraft%2FMCJPG%20on%20branch%20refacter%2Fserverlist%2C%20in%20docs%2F.vitepress%2Ftheme%2Fcomponents%2FServerList.vue%20around%20line%20766%2C%20the%20mediumZoom%20instance%20is%20initialized%20without%20a%20selector.%20After%20fetchServerList%20completes%2C%20add%20a%20nextTick%20call%20to%20attach%20gallery-img%20elements%3A%20after%20await%20fetchServerList()%2C%20add%20await%20nextTick()%3B%20if%20(zoomInstance)%20zoomInstance.attach('.gallery-img')&repo=MineJPGcraft%2FMCJPG)

2. **`document.execCommand('copy')` 已废弃** ([`ServerList.vue:394-398`](https://github.com/MineJPGcraft/MCJPG/blob/refacter/serverlist/docs/.vitepress/theme/components/ServerList.vue#L394))：
   此为 fallback 逻辑，虽然现代浏览器基本都支持 `navigator.clipboard`，但 `execCommand` 已被标记为废弃 API，在某些浏览器的严格模式下可能静默失败。当前实现已在 `try/catch` 中，风险可控，但建议在控制台打印警告提醒。

3. **`nextTick` 已从 `vue` 引入但未使用**：
   ```js
   import { ref, computed, onMounted, onUnmounted, watch, nextTick } from 'vue'
   ```
   `nextTick` 被引入但在代码中未见调用。建议移除以保持代码整洁。[Fix this →](https://claude.ai/code?q=In%20MineJPGcraft%2FMCJPG%20on%20branch%20refacter%2Fserverlist%2C%20in%20docs%2F.vitepress%2Ftheme%2Fcomponents%2FServerList.vue%20line%20325%2C%20remove%20nextTick%20from%20the%20import%20since%20it%20is%20imported%20but%20never%20used&repo=MineJPGcraft%2FMCJPG)

4. **API 端点变更**：主 API 从 `https://serverlist.mcjpg.org/servers.json` 改为 `https://server-editor.mcjpg.dev/api/getjson`，而原 URL 被移到了 fallback 列表。请确认：
   - 新 API 端点的数据格式（尤其 `picture` 字段是否为数组）已通过测试？
   - 新端点是否有 CORS 配置允许来自 `mcjpg.org` 的跨域请求？

#### ⚠️ 代码质量

5. **`paginatedServers` watch 与 `expandedIds` watch 触发的 `attach` 调用重复问题**：
   ```js
   watch([paginatedServers, expandedIds], () => {
     if (zoomInstance) { zoomInstance.attach('.gallery-img') }
   }, { flush: 'post' })
   ```
   每次 `expandedIds` 变化（任意卡片展开/收起），都会对**所有** `.gallery-img` 重新调用 `attach`。`medium-zoom` 的 `attach` 会对已绑定元素产生重复绑定，可能导致图片点击时 zoom 触发多次。建议：在 `attach` 前先 `detach`，或限制只在有画廊图片时绑定。[Fix this →](https://claude.ai/code?q=In%20MineJPGcraft%2FMCJPG%20on%20branch%20refacter%2Fserverlist%2C%20in%20docs%2F.vitepress%2Ftheme%2Fcomponents%2FServerList.vue%2C%20the%20watch%20on%20%5BpaginatedServers%2C%20expandedIds%5D%20calls%20zoomInstance.attach('.gallery-img')%20every%20time%20any%20card%20is%20expanded%20or%20collapsed.%20medium-zoom%20may%20duplicate%20bindings.%20Fix%20by%20calling%20zoomInstance.detach()%20before%20zoomInstance.attach('.gallery-img')%20in%20the%20post-flush%20watcher&repo=MineJPGcraft%2FMCJPG)

6. **`copyTimer` 为模块级变量，非响应式**：
   ```js
   let copyTimer = null
   ```
   这里使用的是普通 `let`（非 `ref`），作为组件级的计时器 ID 存储是合理的，但如果存在多个组件实例同时挂载，可能产生共享问题。当前使用场景下（单例组件）影响不大。

7. **骨架屏中 `.card-header` 与 `.card-expand` 的对应**：骨架屏新增了 `skeleton-line--arrow` 模拟展开箭头，但没有骨架化的展开区域，与实际卡片结构一致，是合理的设计。

#### ✅ 亮点

- **`hasDetails` 标志**设计合理，只有存在可展开内容时才渲染展开箭头和交互，干净。
- **图片加载失败剔除逻辑**（`markPicFailed` + `isPicFailed`）设计得当，避免了显示破损图片。
- **`card-expand` 使用 CSS `grid-template-rows: 0fr → 1fr` 的展开动画**，比 `max-height` hack 更优雅，现代浏览器支持良好。
- **无障碍改进**：`card-header` 改为 `button` 元素，添加了 `aria-expanded`，语义化提升明显。
- **`onUnmounted` 中清理了 `copyTimer` 和 `zoomInstance`**，内存管理到位。

#### 🎨 样式

- 移除了 `transform: translateY(-3px)` hover 效果，交互更稳定，不会因卡片展开后大小变化而产生视觉抖动。
- 新增 `.server-grid` 的 `align-items: start` 避免展开时撑高同行其他卡片，处理得当。
- 文件末尾缺少换行符（`\ No newline at end of file`），建议补充。[Fix this →](https://claude.ai/code?q=In%20MineJPGcraft%2FMCJPG%20on%20branch%20refacter%2Fserverlist%2C%20docs%2F.vitepress%2Ftheme%2Fcomponents%2FServerList.vue%20is%20missing%20a%20newline%20at%20the%20end%20of%20file.%20Please%20add%20a%20trailing%20newline%20after%20the%20closing%20%3C%2Fstyle%3E%20tag&repo=MineJPGcraft%2FMCJPG)

---

### 结论

此 PR 的功能改进方向明确，代码质量整体良好。主要建议在合并前处理：

1. **medium-zoom 初始绑定问题**（可能导致用户无法放大图片）— 优先级高
2. **重复 `attach` 可能导致 zoom 多次触发** — 优先级中
3. **新 API 端点兼容性验证** — 优先级高（需手动确认）
4. **移除未使用的 `nextTick` 导入** — 优先级低
5. **文件末尾补充换行符** — 优先级低

**Review by FireGuo1145 — APPROVED** [review]

**Review by wang-yupu — APPROVED** [review]
感觉像是某种AI generated

**Comment by MCJPG-Agent** [comment]
❤️ 感谢来自 @ZhuYuxuan9302 的 PR ❤️
❤️ Great PR from @ZhuYuxuan9302 ❤️

如果您也对 **MCJPG** 组织的开发工作感兴趣,欢迎联系并加入我们!
If you're also interested in the development work of the **MCJPG** organization, feel free to reach out and join us!
